packages/devtools_app/lib/src/screens/memory/shared/heap/class_filter.dart

```diff
@@ -50,11 +59,30 @@ class ClassFilter {
 
   ClassFilter.empty()
       : this(
-          filterType: ClassFilterType.except,
+          filterType: _defaultFilterType,
           except: defaultExceptString,
           only: null,
         );
 
+  factory ClassFilter.parse(Map<String, dynamic> json) {
+    final type = json[_Json.type] as String?;
+    return ClassFilter(
+      filterType:
+          ClassFilterType.values.lastWhereOrNull((t) => t.name == type) ??
+              _defaultFilterType,
+      except: json[_Json.except] as String? ?? defaultExceptString,
+      only: json[_Json.only] as String?,
+    );
+  }
+
+  Map<String, dynamic> prepareForOffline() {
+    return {
+      _Json.type: filterType.name,
+      _Json.except: except,
+      _Json.only: only,
+    };
```

Comment:
use an extension type for the json parsing in this class. Search "extension type" in the codebase for examples

Reply:
Added TODO.